src/components/Utils/filters.js

```diff
@@ -31,3 +31,83 @@ export const mergeFilters = (userFilter, adminAppliedPreFilter) => {
 
   return filterAB;
 };
+
+function isFilterOptionToBeHidden(option, filtersApplied, fieldName) {
+  if (option.count > 0) {
+    return false;
+  }
+
+  if (typeof filtersApplied !== 'undefined'
+      && filtersApplied[fieldName]
+      && filtersApplied[fieldName].selectedValues.includes(option.key)) {
+    return false;
+  }
+
+  return true;
+}
+
+/**
+   * This function updates the counts in the initial set of tab options
+   * calculated from unfiltered data.
+   * It is used to retain field options in the rendering if
+   * they are still checked but their counts are zero.
+   */
+export const updateCountsInInitialTabsOptions = (
+  initialTabsOptions, processedTabsOptions, filtersApplied,
+) => {
+  const updatedTabsOptions = JSON.parse(JSON.stringify(initialTabsOptions));
```

Comment:
Why do we need to stringify and then parse this object here? 🤔

Reply:
alright it took me time to understand this code block but I guess you were trying to deep clone an object here. :D